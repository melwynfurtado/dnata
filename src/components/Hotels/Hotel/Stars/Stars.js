import React from 'react'
import PropTypes from 'prop-types'

const Stars = ({ stars }) => {
  let starsJsx = []

  for(let i=0; i <5; i++) {
    starsJsx.push(<span key={i} className={`fa fa-star ${stars > i && 'checked'}`}></span>)
  }

  return <div className="stars">{ starsJsx }</div>  
}

Stars.propTypes = {
  stars: PropTypes.number,
}

export default Stars