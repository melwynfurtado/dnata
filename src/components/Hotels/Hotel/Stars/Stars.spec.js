import React from 'react'
import renderer from 'react-test-renderer'
import Stars from './Stars'

describe('Stars Component', () => {
  it('renders correctly for 5 stars', () => {
    const Component = <Stars stars={5} />
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly for 3 stars', () => {
    const Component = <Stars stars={3} />
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })  
})