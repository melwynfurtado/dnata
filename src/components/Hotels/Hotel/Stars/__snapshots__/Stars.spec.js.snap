// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars Component renders correctly for 3 stars 1`] = `
<div
  className="stars"
>
  <span
    className="fa fa-star checked"
  />
  <span
    className="fa fa-star checked"
  />
  <span
    className="fa fa-star checked"
  />
  <span
    className="fa fa-star false"
  />
  <span
    className="fa fa-star false"
  />
</div>
`;

exports[`Stars Component renders correctly for 5 stars 1`] = `
<div
  className="stars"
>
  <span
    className="fa fa-star checked"
  />
  <span
    className="fa fa-star checked"
  />
  <span
    className="fa fa-star checked"
  />
  <span
    className="fa fa-star checked"
  />
  <span
    className="fa fa-star checked"
  />
</div>
`;
